middleware/capacity_test.go

```diff
@@ -48,25 +102,44 @@ func TestItErrorsWhenNoJobCapacityAvailable(t *testing.T) {
 	coordinator := pipeline.NewStubCoordinatorOpts(pipeline.StrategyCatalystFfmpegDominance, nil, pipeFfmpeg, nil)
 	coordinator.InputCopy = &clients.StubInputCopy{}
 
-	// Create a lot of in-flight jobs
-	for x := 0; x < 8; x++ {
+	// Set up the HTTP handler
+	cm := CapacityMiddleware{}
+	handler := cm.HasCapacity(coordinator, next)
+	responseRecorder := httptest.NewRecorder()
+
+	// Create a lot of in-flight regular-vod jobs
+	for x := 0; x < config.MaxInFlightJobs; x++ {
 		coordinator.StartUploadJob(pipeline.UploadJobPayload{
 			RequestID: fmt.Sprintf("request-%d", x),
 		})
 	}
 	time.Sleep(1 * time.Second)
-
-	// Set up the HTTP handler
-	cm := CapacityMiddleware{}
-	handler := cm.HasCapacity(coordinator, next)
-
 	// Call the handler
-	responseRecorder := httptest.NewRecorder()
-	handler(responseRecorder, nil, nil)
-
+	handler(responseRecorder, vodReq, nil)
 	// Confirm we got an HTTP 429 response
 	require.Equal(t, http.StatusTooManyRequests, responseRecorder.Code)
+	// Confirm the handler didn't call the next middleware
```

Comment:
Let's split the clipping failure mode out into a new test so that it's clearer what the cause is when a test fails

Reply:
Done